core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -485,4 +485,13 @@ package object config {
         "array in the sorter.")
       .intConf
       .createWithDefault(Integer.MAX_VALUE)
+
+  private[spark] val SHUFFLE_MAP_OUTPUT_STATISTICS_PARALLEL_AGGREGATION_THRESHOLD =
+    ConfigBuilder("spark.shuffle.mapOutputStatistics.parallelAggregationThreshold")
+      .internal()
+      .doc("Multi-thread is used when the number of mappers * shuffle partitions exceeds this " +
+        "threshold")
+      .intConf
+      .createWithDefault(100000000)
```

Comment:
wow 100 million is really a large threshold, how do you pick this number?

Reply:
Now I also think it's a little bit large... In the case I mentioned, the 5s gap is created by 10^8 of this value. Maybe 10^7 is good?